### PR TITLE
🏄‍♂️ Reusable include for rendering dependent object lists

### PR DESCRIPTION
**Pull request by rowanseymour**
Still haven't converted fields and groups over to be soft dependencies because they're a bit more complicated in that they can be dependencies of other things. This is a step toward having easier ways of dealing with different dependent objects consistently..

Before 

Captura de Pantalla 2021-07-14 a la(s) 17 11 50
Captura de Pantalla 2021-07-14 a la(s) 17 13 50
Captura de Pantalla 2021-07-14 a la(s) 17 12 05

After 

Captura de Pantalla 2021-07-14 a la(s) 17 12 57
Captura de Pantalla 2021-07-14 a la(s) 17 13 06
Captura de Pantalla 2021-07-14 a la(s) 17 13 14

And also re-used on the delete modal shared by a bunch of types, e.g.

Captura de Pantalla 2021-07-14 a la(s) 17 15 47

### CODE REVIEW

File: temba/campaigns/models.py [hunk 1]
```diff
@@ -404,6 +408,10 @@ def get_hour_choices(cls):
             hours.append((i, "at %s:00 %s" % (hour, period)))
         return hours
 
+    @property
+    def name(self):
+        return f"{self.campaign.name} ({self.offset_display} {self.relative_to.name})"
```
Comment (rowanseymour) [inline temba/campaigns/models.py:413]: life is easier when everything has a `name` property...

**Comment by ericnewcomer** [comment]
One thought with switching to soft dependencies is we probably can't get away with bundling things with "2 more.." etc. This was fine when each one needed to be addressed, but now we probably need to allow them to review each one as they may not remove the dependency.

**Comment by rowanseymour** [comment]
Tho our hands are a bit tied with these being in modals. What if tho "and X more" linked to a page which listed all the dependencies? Could that work generically across different types?

**Comment by ericnewcomer** [comment]
That is was actually my first thought, but a dedicated page for this also feels a bit weird and overkill. But ya, there might not be a way around that.

**Comment by rowanseymour** [comment]
@ericnewcomer in the delete modal case this PR doesn't change anything for most dependency types - those are currently limited to 5... bump that to 10? Proceed with this with a view to adding a separate dependency listing page at some point?

**Comment by ericnewcomer** [comment]
Ya, I mean I think I'm okay proceeding with the max of 5 for now. I just wanted to call out a way the change might affect users.

**Review by ericnewcomer — APPROVED** [review]

**Pull request merged**